packages/carequality-sdk/src/client/carequality-fhir.ts

```diff
@@ -0,0 +1,201 @@
+      },
+      retryCondition: error => {
+        return error.response?.status !== 200 && error.response?.status !== 403; // As per the Carequality Implementation Guide: https://carequality.org/healthcare-directory/operational_best_practices.html#reliability
+      },
+    });
+
+    this.apiKey = apiKey;
+  }
+
+  buildHeaders(headers?: Record<string, string>): Record<string, string> {
+    return {
+      ...headers,
+      "Accept-Encoding": "gzip",
+    };
+  }
+
+  async sendGetRequest(url: string, headers?: Record<string, string>): Promise<AxiosResponse> {
+    return this.api.get(url, { headers: this.buildHeaders(headers) });
+  }
+
+  async sendPostRequest(
+    url: string,
+    data: unknown,
+    headers?: Record<string, string>
+  ): Promise<AxiosResponse> {
+    return this.api.post(url, data, { headers: this.buildHeaders(headers) });
+  }
+
+  async sendPutRequest(
+    url: string,
+    data: unknown,
+    headers?: Record<string, string>
+  ): Promise<AxiosResponse> {
+    return this.api.put(url, data, { headers: this.buildHeaders(headers) });
+  }
+
+  /**
+   * Lists the indicated number of organizations.
+   *
+   * @param count Optional, number of organizations to fetch. Defaults to 1000.
+   * @param start Optional, the index of the directory to start querying from. Defaults to 0.
+   * @param oid Optional, the OID of the organization to fetch.
+   * @param active Optional, indicates whether to list active or inactive organizations. Defaults to true.
+   * @returns
+   */
+  async listOrganizations({
+    count = MAX_COUNT,
+    start = 0,
+    oid,
+    active,
+  }: ListOrganizationsParams): Promise<Organization[]> {
+    if (count < 1 || count > MAX_COUNT) {
+      throw new Error(`Count value must be between 1 and ${MAX_COUNT}`);
+    }
+
+    const query = new URLSearchParams();
+    query.append("apikey", this.apiKey);
+    query.append("_count", count.toString());
+    start !== undefined && query.append("_start", start.toString());
+    oid !== undefined && query.append("_id", oid);
+    active !== undefined && query.append("active", active.toString());
+
+    const url = `${CarequalityManagementAPIImplFhir.ORG_ENDPOINT}?${query.toString()}`;
+    const resp = await this.sendGetRequest(url, { "Content-Type": "application/json" });
+    const bundle = resp.data as Bundle;
+    return (bundle.entry ?? []).map(e => e.resource as Organization);
+  }
+
+  /**
+   * Registers an organization with the Carequality directory.
+   *
+   * @param org string containing the organization resource (in XML format)
+   * @returns an XML string containing an OperationOutcome resource - see Carequality documentation for details - https://carequality.org/healthcare-directory/OperationOutcome-create-success-example2.xml.html
+   */
+  async registerOrganization(org: Organization): Promise<Organization> {
+    const query = new URLSearchParams();
+    query.append("apikey", this.apiKey);
```

Comment:
This is the only way to send the API Key? This is not the best practice, as URLs and parameters get logged. If we can, we should prob send that on the header.